rclcpp/src/rclcpp/parameter.cpp

```diff
@@ -23,49 +23,54 @@
 using rclcpp::parameter::ParameterType;
 using rclcpp::parameter::ParameterVariant;
 
-ParameterVariant::ParameterVariant()
-: name_("")
+ParameterVariant::ParameterVariant(const std::string & name)
+: name_(name)
 {
   value_.type = rcl_interfaces::msg::ParameterType::PARAMETER_NOT_SET;
+  value_.bool_value = false;
+  value_.integer_value = 0;
+  value_.double_value = 0.0;
+  value_.string_value = "";
+  value_.bytes_value = std::vector<uint8_t>();
 }
 
 ParameterVariant::ParameterVariant(const std::string & name, const bool bool_value)
-: name_(name)
+: ParameterVariant(name)
```

Comment:
If we don't initialize the unused value fields this as well as the added optional argument to the default constructor wouldn't be necessary.

Same below.